Automattic/vip-go-mu-plugins
# Tweak Jetpack Connection Pilot to autoconnect

## Description

This PR aims to run Jetpack Connection Pilot logic earlier so we can automatically connect newly created sites. We also hook to `wp_initialize_site` and `wp_update_site` to immediately run the same logic whenever changes (or creation) happens on the site.

We also make sure that reconnection is only attempted on production sites.

## Changelog Description

### Jetpack Connection Pilot

Improved automatic Jetpack connection process.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1. Check out PR.
2. Set up a new site (or a subsite in a network) on a production environment.
3. Make sure that Jetpack is initially disconnected.
4. In a short time, you should see Jetpack enabled.
